### PR TITLE
Transactions specs

### PR DESCRIPTION
**Pull request by vogdb**

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review

File: test/transaction.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      done();
+    });
+  });
+
+  it('transaction callback returns promise', function(done) {
+    var transaction = new Transaction('id', _.constant(Promise.resolve('resolved')));
+    transaction.execute().then(function(result) {
+      assert.equal(result, 'resolved');
+      done();
+    });
```
Comment (tomaszdurka) [inline test/transaction.js]: Add `.catch(done)`
- by pain in the ass...

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by tomaszdurka** [comment]
lgtm

**Pull request merged**